src/components/LocationForm/index.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+
+import type {
+  FormChildProps,
+  FormFieldChildProps,
+} from '../../common/form/types';
+import type { StateConfig } from './stateList';
+
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import { StyleSheet, View } from 'react-native';
+import { Button, FormValidationMessage } from 'react-native-elements';
+import STATE_LIST from './stateList';
+import Form from '../../common/form/Form';
+import FormField from '../../common/form/FormField';
+
+import TextField from '../TextField';
+import Picker from '../Picker';
+
+const REQUIRED_FIELDS = [
+  'city',
+  'locationType',
+  'name',
+  'state',
+  'street',
+  'zipCode',
+];
+
+const isRequiredMessage = (fieldName: string): string =>
+  `${fieldName} is required`;
+
+const validate = (values: Location): { [key: string]: string } => {
+  const errors = {};
+
+  REQUIRED_FIELDS.forEach((fieldName: string) => {
+    if (!values[fieldName]) {
+      errors[fieldName] = isRequiredMessage(fieldName);
+    }
+  });
+
+  return errors;
+};
+
+const styles = StyleSheet.create({
+  formContainer: {
+    paddingBottom: 15,
+  },
+});
+
+type Props = {|
+  onSubmit: (values: Location) => Promise<void>,
+  submitButtonLabel: string,
+|};
+
+// todo figure out and implement good pattern for handling
+// fields focus order
```

Comment:
Is there a tab-index field like on the web?

Reply:
I found only manual solutions like: onSubmitText={() => this.selectNextInput}` where this.selectNextInput, call setFocus() on the next input ref of inputs refs stack.